x/oracle/keeper/abci.go

```diff
@@ -0,0 +1,36 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// BeginBlocker is called at the beginning of every block.  It resets the count of
+// removed currency pairs.
+func (k *Keeper) BeginBlocker(goCtx context.Context) error {
+	// unwrap the context
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	removes, err := k.numRemoves.Get(ctx)
+	if err != nil {
+		return err
+	}
+
+	numCPs, err := k.numCPs.Get(ctx)
```

Comment:
So semantically, `numCPs - numRemoves` is the actual # of currency-pairs in state at any point in time? `numRemoves` points to the latest state, while `numCPs` points to the `latestState - 1`?

Reply:
correct.  Do you think it would be better to rename the variables ?